### DISCUSSION TITLE
Include Windows Installer?

### DISCUSSION THREAD

**Discussion by kaikreuzer** [openhab/openhab-distro#1491] [Architecture Council Discussions]
Dear colleagues,

I just want to briefly point you to https://github.com/openhab/openhab-distro/issues/807.
Would you be ok to make this an official part of openHAB by moving the repo to our org?
Do we have a chance to have it maintained and build and is it valuable for our community?

I think Tim put a lot of effort in it and for Windows users it can be indeed quite helpful (similar to openHABian for Linux users). I am just a bit uncertain about its build process, which could cause us some trouble, but I am not sure.

Let me know if you think we should NOT do it - then I will try to halt the efforts and rather keep it as an external tool. Otherwise, I'll move on and see what needs to be done to move it to openHAB.

**Comment by ghys** [comment]
I am giving it a try and it seems to be very well done.
Note that (probably) due to the lack of signature on the executable, it is blocked from downloading and running by Windows (and the browser).

![image](https://user-images.githubusercontent.com/2004147/59253470-981e0800-8c2e-11e9-8ecf-d5d2bca8b20c.png)
(the Run Anyway option is not even shown intially, we have to click on a "More info" link to reveal it).
If a paid certificate is necessary, supplying it and publishing a signed version of the executable on the website could probably be taken care of by the Foundation - similar to e.g. the publishing of the mobile apps.

After that the installation went smoothly, it seems to be able to schedule update and backup jobs, and also offered to install as a service, but I didn't try that. It is also able to install any version back to 2.0 (which is unfortunately selected by default), that's probably unnecessary.

I think it's very nice and would be a valuable addition to the project.

**Comment by wborn** [comment]
A good Windows installer would certainly be a nice addition to the project. Though it seems we already list [Chocolatey](https://chocolatey.org/packages/openhab) as an installer on the Windows [download page](https://www.openhab.org/download/). Preferably I'd see the (Windows) openHAB community to concentrate their efforts on providing one very good Windows installer.

**Comment by wborn** [comment]
>  I am just a bit uncertain about its build process, which could cause us some trouble, but I am not sure.

There seems to be a somewhat beta Travis CI [Windows Build Environment](https://docs.travis-ci.com/user/reference/windows/) nowadays. If it works well enough, we could trigger it just like how we trigger Travis for building the Docker images.
